REFERENCE.md

```diff
@@ -650,8 +650,9 @@ Fails a compile if the client system is not compatible with the module's
 `metadata.json`
 
 #### `simplib::assert_metadata(String[1] $module_name, Optional[Struct[{
```

Comment:
:thinking:  Thinking to how this function could be used after this feature merges:  different portions of a given module might not support all OSes equally, so the function's `$module_name` parameter might be better off re-named as something more generic, like `$section`, `$identifier`, `$subject`, or `$name_of_thing_that_will_throw_a_fit_if_it_doesnt_like_your_os`.

Reply:
@op-ct This is literally the module that you have to go look in and is the `$module_name` variable in Puppet. I figured that this was the best hint that I could give :-/.

I guess we could add a 'notes' section at some point but that's getting pretty busy.

Reply:
> This is literally the module that you have to go look in and is the $module_name variable in Puppet

@trevor-vaughan that is literally true, but come simp/pupmod-simp-simp#217, the message _`OS 'Windows' version 'Server 2012 R2' is not supported by module 'simp'`_ will be literally *un*true—the simp module does support Windows (in its scenario classes).  When a user follows up that error by looking in the module, its `metadata.json` will tell them the (literally) opposite story.

The misleading error message is really the problem, so we should at least alter the failure text to mention when the assertion fails due to blacklisting.  That was , users will be cued to expect support in other areas of the module.